airflow/www/static/js/dag/details/taskInstance/index.tsx

```diff
@@ -90,6 +92,8 @@ const TaskInstance = ({
 
   if (!group || !run || !instance) return null;
 
+  const updateTaskInstanceNotesCallable = (newNotes: string) => { instance.notes = newNotes; };
```

Comment:
We shouldn't mutate instance like this.

Let's keep notes as a state variable with `const [notes, setNotes] = useState(instance.notes || '');`

And then here we can call `setNotes(newNotes);`

Reply:
oh wait, you do that in `SetDagTaskNotes.tsx`. What is this doing?

Reply:
Yes so the idea in both cases is that we update the data we fetched from the API.
The reason for that is that when we switch DagRuns/TaskInstance views and go back, that it will still show you the most up to date task note. I figured we could either reload the entire dataset, or we could simply update the data we fetched, so I went for the second option.
How would you approach this issue?

Reply:
I very much like that idea. Much less hacky.
 Let me give that a try :)

Reply:
You can look at the [docs](https://react-query-v3.tanstack.com/guides/optimistic-updates) for optimistic updates for an example of how to get the query data for a different query and then update it.